projects/dotcms-field-elements/src/components/dot-textfield/dot-texfield.e2e.ts

```diff
@@ -1,140 +1,330 @@
+                expect(await dotTestUtil.getHint(page)).toBeNull();
+            });
+
+            it('should not render  hint with invalid value', async () => {
+                element.setProperty('hint', { test: 'hint' });
+                await page.waitForChanges();
+                expect(await dotTestUtil.getHint(page)).toBeNull();
+            });
+        });
+
+        describe('required', () => {
+            it('should render required attribute and dot-required class', async () => {
+                element.setProperty('required', 'true');
+                await page.waitForChanges();
+                expect(input.getAttribute('required')).toBeDefined();
+                expect(element).toHaveClasses(['dot-required']);
+            });
+
+            it('should not render required attribute', async () => {
+                element.setProperty('required', 'false');
+                await page.waitForChanges();
+                expect(input.getAttribute('required')).toBeNull();
+            });
+
+            it('should render required attribute with invalid value', async () => {
+                element.setProperty('required', { test: 'test' });
+                await page.waitForChanges();
+                expect(input.getAttribute('required')).toBeDefined();
+            });
+        });
+
+        describe('requiredMessage', () => {
+            it('should render requiredMessage', async () => {
+                element.setProperty('required', 'true');
+                element.setProperty('requiredMessage', 'Test');
+                await input.press('a');
+                await input.press('Backspace');
+                await page.waitForChanges();
+                expect((await dotTestUtil.getErrorMessage(page)).innerText).toBe('Test');
+            });
+
+            it('should not render requiredMessage', async () => {
+                await page.waitForChanges();
+                expect(await dotTestUtil.getErrorMessage(page)).toBe(null);
+            });
+
+            it('should not render and not break with with invalid value', async () => {
+                element.setProperty('required', 'true');
+                element.setProperty('requiredMessage', { test: 'hi' });
+                await input.press('a');
+                await input.press('Backspace');
+                await page.waitForChanges();
+                expect(await dotTestUtil.getErrorMessage(page)).toBeNull();
+            });
+        });
+
+        describe('regexCheck', () => {
+            it('should set empty value when invalid regexCheck', async () => {
+                element.setProperty('regexCheck', '[*');
+                await page.waitForChanges();
+                expect(await element.getProperty('regexCheck')).toBe('');
+                expect(await dotTestUtil.getErrorMessage(page)).toBeNull();
+            });
+        });
+
+        describe('validationMessage', () => {
+            it('should render validationMessage', async () => {
```

Comment:
We need to test the default validation message.

Reply:
'should render validationMessage when value doesn't match with regexCheck '